### ISSUE TITLE
Exception during ADO ingest queue item processing causes the queue to be stuck.

### ISSUE THREAD

**Issue by patdunlavey** [esmero/ami#221]
If an exception is encountered during `ami_ingest_ado` queue processing (specifically, inside the "[try](https://github.com/esmero/ami/blob/main/src/AmiBatchQueue.php#L53-L86)-catch" section) in [`\Drupal\ami\AmiBatchQueue::takeOne`](https://github.com/esmero/ami/blob/0.8.0/src/AmiBatchQueue.php), the problematic queue item is not removed from the queue and is picked up again to be processed again. This causes the queue to become permanently stuck on that item if the exception repeats every time the queue processor attempts to process it. The only solution is to manually go in (assuming you're using drupal database for queue management and not redis) and manually delete the offending queue item.

We've seen an AMI ingest queue get stuck for several days before anybody notices (e.g. over a holiday weekend)! We've seen AMI update queues repeatedly generate new revisions for objects over and over, hundreds of times. (It would seem that the queue item is hitting an exception *after* the ADO has been updated.)

If the exception type is "RequeueException" or "SuspendQueueException", the lease is released, making the item become immediately first in line to be processed again. Other exceptions simply keep the item leased until the lease expires, and then they're ready to process it again. 

Perhaps this makes sense for "RequeueException" - assuming that whichever place threw the exception has a valid reason for determining that the item should be re-enqueued (so hard to follow in the code!). I can't see the logic for keeping an item in the queue for any other kind of exception.

Thoughts:
- For any kind of exception other than "RequeueException", I think we could just delete the queue item and log an error.
- For "RequeueException", I'm not sure. I thought of deleting the queue item and re-enqueueing it at the end of the queue, but...
- Both of these ideas could cause problems if later queue items depend on the failed item - e.g. a parent item fails and then is followed by a child. Can we depend on the child creation item to fail if the parent item does?
- Exceptions can be transient - not usually, but they *can* be. But I don't think it makes sense to have a "re-try" limit (of, say, three tries). Best, I think, to fail and do a really good job of logging the exception.

**Comment by DiegoPino** [comment]
@patdunlavey could you share what generated the exception?

**Comment by patdunlavey** [comment]
I really am not sure! Initial thought was bad character encoding, but I actually think it was strings in the csv that were not intended as json encoded, but were decoded as valid json.

**Comment by DiegoPino** [comment]
Solved via https://github.com/esmero/ami/commit/f8e42acf7b79983a4babe86ed50f46f413735e36
